File: LeetCode/py/832.py
#Link: https://leetcode.com/problems/flipping-an-image/

class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        n = len(image)
        m = (n//2) if (n%2==0) else (n//2 + 1)
        for i in range(n):
            for j in range(m):
                a = int(not image[i][j])
                b = int(not image[i][n-j-1])
                image[i][j] = b
                image[i][n-j-1] = a
        return image
        

File: LeetCode/cpp/832.cpp
//Link: https://leetcode.com/problems/flipping-an-image/

class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& image) {
        int n = image.size(), a, b;
        int m = (n%2==0) ? n/2 : n/2 + 1;
        for(int i = 0 ; i < n ; i++) {
            for(int j = 0 ; j < m ; j++) {
                a = !image[i][j];
                b = !image[i][n-j-1];
                image[i][j] = b;
                image[i][n-j-1] = a;
            }
        }
        return image;
    }
};
